### PR TITLE
Fixed unhandled case where an entity with tags got all his tags removed.

### LINKED ISSUE #1: Not possible to remove all tags
If I have an entity with at least one tag and I try to remove all tags (so it has no more tags), the change will not be persisted.

As a removal of all tags is registered as an update event as well (due to the updatedAt), the important part will be skipped because $tagNames will be NULL and skipped at:

```
    protected function setTags(TaggableInterface $entity, $update = false)
    {
        ...
        if (empty($tagNames)) {
            return;
        }
        ...
    }
```

This makes it impossible to remove all tags from an entity.

### PR DESCRIPTION
**Pull request by adrianrudnik**
Fixes #1.

### CODE REVIEW

File: Listener/TagSubscriber.php [hunk 1]
```diff
@@ -98,33 +98,37 @@ public function onFlush(OnFlushEventArgs $args)
     protected function setTags(TaggableInterface $entity, $update = false)
     {
         $tagNames = $entity->getTagNames();
-        if (empty($tagNames)) {
-            return;
-        }
+
         // need to clone here, to avoid getting new tags
-        $oldTags = is_object($entityTags = $entity->getTags()) ? clone $entityTags : $entityTags;
+        $oldTags          = is_object($entityTags = $entity->getTags()) ? clone $entityTags : $entityTags;
```
Comment (garak) [inline Listener/TagSubscriber.php]: Please unalign equal signs.

**Comment by garak** [comment]
Can you fix failing tests? And then squash your commits.
Thanks

**Comment by adrianrudnik** [comment]
Sure, can do it early next week, busy right now.

**Comment by garak** [comment]
Great, thanks!

**Comment by garak** [comment]
@kreischweide any news about this?

**Comment by garak** [comment]
Fixed in d0ba024a7aecf6b9c5ff258b8499f56808f40d8a

**Pull request closed without merging**